src/components/TextArea/TextArea.tsx

```diff
@@ -28,40 +34,70 @@ export type TextAreaProps = {
   onKeyDown?: React.KeyboardEventHandler<HTMLTextAreaElement | HTMLInputElement>;
   /** Callback fired when the `input` value typed is changed */
   onInput?: React.EventHandler<any>;
-} & Pick<TextInputBaseProps, 'status'> &
+} & Pick<HTMLTextAreaElement, 'cols' | 'rows'> &
```

Comment:
i dont like that we pick some of the input and some of the textarea props. As a user I might want to pass any textarea prop.